### ISSUE TITLE
'devtools' websocket?

### ISSUE THREAD

**Issue by aleclarson** [facebook/react-native#8674]
What is the `devtools` endpoint used for?

**Comment by aleclarson** [comment]
> This means the new executorClass may have its setUp method called before the invalidation is completed.

This probably doesn't matter, because the `RCTBridge` creates a new `RCTBatchedBridge` that doesn't care if the invalidation occurs asynchronously. Is that right?

**Comment by aleclarson** [comment]
Actually, hot reloading being enabled isn't relevant, because the `devtools` websocket is used no matter what (as seen in [`InitializeJavaScriptAppEngine.js`](https://github.com/facebook/react-native/blob/22fbb6d46d4b91bfe08b83aaba46ad92a32bf984/Libraries/JavaScriptAppEngine/Initialization/InitializeJavaScriptAppEngine.js#L233)).

With that information, I think I can assume that `devtools` is the websocket used by the native iOS app to talk to the packager; while `debugger-proxy` is what the Chrome debugger uses to talk to the packager.

Does that all sound right? 😃

**Comment by aleclarson** [comment]
Okay, I only thought hot reloading was relevant for one reason: the `debugger-proxy` websocket was being invalidated when hot reloading was enabled. That only happens because the `[RCTBridge reload]` was being called a **second** time by [`[RCTDevMenu setExecutorClass:]`](); while the first call is of course from [`[RCTDevMenu setHotLoadingEnabled:]`](https://github.com/facebook/react-native/blob/master/React/Modules/RCTDevMenu.m#L612)).

I figured we might as well avoid that second reload.
So I added `BOOL _reloading` to the `RCTBridge`, and set it like below:

```objc
- (void)reload
{
  /**
   * Any thread
   */
  dispatch_async(dispatch_get_main_queue(), ^{
    if (_reloading) {
      return;
    }
    _reloading = YES;
    [self invalidate];
    [self setUp];
    _reloading = NO;
  });
}
```

This prevents the second reload successfully. Perfect, now I understand that `debugger-proxy` is always used (even with hot reloading enabled).

Now the question is: "What's the point of the `devtools` websocket?"
